src/ring_theory/localization/away.lean

```diff
@@ -194,3 +196,69 @@ lemma is_localization.away.finite_presentation (r : R) {S} [comm_ring S] [algebr
   [is_localization.away r S] : algebra.finite_presentation R S :=
 (adjoin_root.finite_presentation _).equiv $ (localization.away_equiv_adjoin r).symm.trans $
   is_localization.alg_equiv (submonoid.powers r) _ _
+
+section num_denom
+
+open multiplicity unique_factorization_monoid is_localization
+
+variable (x : R)
+
+variables (B : Type*) [comm_ring B] [algebra R B] [is_localization.away x B]
+variable (hx : irreducible x)
+
+/-- `self_as_unit` is the element `x` at which `B` is a localization away, bundled as a unit in `B`
+-/
+noncomputable def self_as_unit : Bˣ := ⟨algebra_map _ _ x, away.inv_self x, away.mul_inv_self _,
+    by {rw mul_comm, exact away.mul_inv_self _}⟩
+
+lemma self_as_unit_pow_sub (a : R) (b : B) (m d : ℤ) :
+  (((self_as_unit x B ^ (m - d)) : Bˣ) : B) * mk' B a (1 : submonoid.powers x) = b ↔
+  (((self_as_unit x B ^ m) : Bˣ) : B) * mk' B a (1 : submonoid.powers x) =
+    (((self_as_unit x B ^ d) : Bˣ) : B) * b :=
+by {simp only [zpow_sub, units.coe_mul, mul_comm (((self_as_unit x B ^ m) : Bˣ) : B) _,  mul_assoc,
+  units.inv_mul_eq_iff_eq_mul]}
+
+lemma pow_eq_algebra_map (d : ℕ) : (((self_as_unit x B)^(d : ℤ) : Bˣ) : B) =
+  (algebra_map R B x)^d := by {simp only [self_as_unit, zpow_coe_nat, units.coe_pow, units.coe_mk]}
+
+variables [is_domain R] [normalization_monoid R] [unique_factorization_monoid R]
+include hx
+
+lemma max_power_factor {a₀ : R} (h : a₀ ≠ 0) [nontrivial R] :
```

Comment:
Can you move this to `unique_factorization_domain.lean`? (Or perhaps add a remark to do so after the port?)

Reply:
Do you mean just this lemma or all the additions?

Reply:
I'll try later.

Reply:
I have moved `max_power_factor` to `unique _factorization_domain` and I have opened the corresponding PR in `mathlib4`.